### PR TITLE
Be able to set custom error message from rspec-expectation

### PR DESCRIPTION
**Pull request by benoittgt**
Close: https://github.com/rspec/rspec-mocks/issues/1250

### CODE REVIEW

File: features/basics/expecting_messages.feature [hunk 1]
```diff
@@ -73,3 +73,40 @@ Feature: Expecting messages
       """
      When I run `rspec negative_message_expectation_spec.rb`
      Then the examples should all pass
+
+  Scenario: Failing positive message expectation with custom failure message
```
Comment (JonRowe) [inline features/basics/expecting_messages.feature]:
```suggestion
  Scenario: Failing positive message expectation with a custom failure message
```

File: features/basics/expecting_messages.feature [hunk 2]
```diff
@@ -73,3 +73,40 @@ Feature: Expecting messages
...
+    """
+    When I run `rspec example_spec.rb --format documentation`
+    Then the output should contain:
+    """
+      1) An unfulfilled positive message expectation triggers a failure
+         Failure/Error: expect(dbl).to receive(:foo), "dbl never call :foo"
+           dbl never call :foo
+    """
+
+  Scenario: Failing negative message expectation with custom failure message
```
Comment (JonRowe) [inline features/basics/expecting_messages.feature]:
```suggestion
  Scenario: Failing negative message expectation with a custom failure message
```

File: features/basics/expecting_messages.feature [hunk 3]
```diff
@@ -73,3 +73,40 @@ Feature: Expecting messages
...
+    Given a file named "example_spec.rb" with:
+    """ruby
+    RSpec.describe "A negative message expectation" do
+      it "fails when the message is received" do
+        dbl = double
+        expect(dbl).not_to receive(:foo), "dbl called :foo but should not"
```
Comment (JonRowe) [inline features/basics/expecting_messages.feature]:
```suggestion
        expect(dbl).not_to receive(:foo), "dbl called :foo but is not supposed to"
```

File: features/basics/expecting_messages.feature [hunk 4]
```diff
@@ -73,3 +73,40 @@ Feature: Expecting messages
...
+        dbl.foo
+      end
+    end
+    """
+    When I run `rspec example_spec.rb --format documentation`
+    Then the output should contain:
+    """
+      1) A negative message expectation fails when the message is received
+         Failure/Error: dbl.foo
+           dbl called :foo but should not
```
Comment (JonRowe) [inline features/basics/expecting_messages.feature]:
```suggestion
           dbl called :foo but is not supposed to
```

**Comment by benoittgt** [comment]
Thanks Jon! Waiting for the CI, then I will merge.

File: features/basics/expecting_messages.feature [hunk 5]
```diff
@@ -73,3 +73,40 @@ Feature: Expecting messages
...
      When I run `rspec negative_message_expectation_spec.rb`
      Then the examples should all pass
+
+  Scenario: Failing positive message expectation with a custom failure message
+    Given a file named "example_spec.rb" with:
+    """ruby
+    RSpec.describe "An unfulfilled positive message expectation" do
+      it "triggers a failure" do
+        dbl = double
+        expect(dbl).to receive(:foo), "dbl never call :foo"
```
Comment (JonRowe) [inline features/basics/expecting_messages.feature]:
```suggestion
        expect(dbl).to receive(:foo), "dbl never calls :foo"
```

File: features/basics/expecting_messages.feature [hunk 6]
```diff
@@ -73,3 +73,40 @@ Feature: Expecting messages
...
+      end
+    end
+    """
+    When I run `rspec example_spec.rb --format documentation`
+    Then the output should contain:
+    """
+      1) An unfulfilled positive message expectation triggers a failure
+         Failure/Error: expect(dbl).to receive(:foo), "dbl never call :foo"
```
Comment (JonRowe) [inline features/basics/expecting_messages.feature]:
```suggestion
         Failure/Error: expect(dbl).to receive(:foo), "dbl never calls :foo"
```

File: features/basics/expecting_messages.feature [hunk 7]
```diff
@@ -73,3 +73,40 @@ Feature: Expecting messages
...
+           dbl never call :foo
```
Comment (JonRowe) [inline features/basics/expecting_messages.feature]:
```suggestion
           dbl never calls :foo
```

**Comment by JonRowe** [comment]
> Thanks Jon! Waiting for the CI, then I will merge.

I just tweaked grammar again (used the magic Github UI to commit how good are suggestions!)

> Edit: Do we want to add a changelog for that?
Yes

**Pull request merged**